app/models/recent_activity.rb

```diff
@@ -69,6 +69,12 @@ def self.user_activity(user, reject_keys=[])
   # All activities that are public and should be visible for a user
   # * +user+ - the user which activities will be returned
   def self.user_public_activity user
```

Comment:
The new version of public activity actually has a field named 'recipient' (we have it in the DB) and maybe that's what it is for. The user is the recipient of those activities, as in he is the one that has them on his activity timeline.